src/classes/Game.js

```diff
@@ -0,0 +1,137 @@
+    this.snakeSpeed = SNAKE_SPEED;
+  }
+
+  getRandomFoodType() {
+    const food = Math.random() < 0.7 ? new Cherry(this.snake) : new Banana(this.snake);
+    return food;
+  }
+
+  main(currentTime) {
+    if (this.isGameOver) {
+      const username = this.username;
+      const points = this.snake.points;
+      const time = this.time;
+      this.saveScore(username, points, time);
+
+      if (confirm(`${username} lost. Points: ${points} Click OK to restart. ${time}`)) {
+        window.location = '';
+      }
+      return;
+    }
+
+    window.requestAnimationFrame(this.main.bind(this));
+    const secondsSinceLastRender = (currentTime - this.lastRenderTime) / 1000;
+    if (secondsSinceLastRender < 1 / this.snakeSpeed) return;
+
+    this.lastRenderTime = currentTime;
+
+    this.update();
+    this.render();
+  }
+
+  update() {
+    const inputDirection = this.input.getInputDirection();
+    this.snake.movement(inputDirection);
+
+    if (this.snake.onSnake(this.food.food)) {
+      this.snake.grow(this.food);
+      this.food = this.getRandomFoodType();
+      this.snakeSpeed += 0.5;
+    }
+
+    this.checkForGameOver();
+  }
+
+  render() {
+    this.gameBoard.innerHTML = '';
+    this.snake.render(this.gameBoard);
+    this.food.render(this.gameBoard, this.food.frType);
+  }
+
+  checkForGameOver() {
+    this.isGameOver = this.snake.isIntersecting();
+  }
+
+  start() {
+    let username = prompt('Enter username: ');
+
+    while (!username) {
+      username = prompt('Username cannot be null. Please enter a username: ');
+    }
+    this.username = username;
+
+    const currentStartTime = DateTime.now().toLocaleString();
+    this.time = currentStartTime;
+    this.displayScoreboard();
+    window.requestAnimationFrame(this.main.bind(this));
+  }
+
+  saveScore(username, points, time) {
+    const scores = JSON.parse(localStorage.getItem('scores')) || [];
+    scores.push({ username, points, time });
+    localStorage.setItem('scores', JSON.stringify(scores));
+  }
+
+  getScores() {
+    return JSON.parse(localStorage.getItem('scores')) || [];
+  }
+
+  displayScoreboard() {
+    const scoreList = document.getElementById('score-list');
+    scoreList.innerHTML = '';
+    const scores = this.getScores();
+
+    scores.sort((a, b) => {
+      if (b.points === a.points) {
+        return b.time - a.time;
+      }
+      return b.points - a.points;
+    });
+
+    scores.forEach((score, index) => {
+      const scoreItem = document.createElement('li');
+      scoreItem.textContent = `${index + 1}. ${score.username}: ${score.points} points. ${this.time}`;
+
+      const deleteButton = document.createElement('button');
+      deleteButton.textContent = 'Delete';
+      deleteButton.addEventListener('click', () => this.removeScore(index));
+
+      scoreItem.appendChild(deleteButton);
+      scoreItem.classList.add('score-item');
+      scoreList.appendChild(scoreItem);
+    });
+  }
+
+  removeScore(index) {
```

Comment:
you can separate this in separate classes that is will interact with local storage